reframe/frontend/statistics.py

```diff
@@ -84,6 +84,10 @@ def failure_report(self):
             report.append('  * System partition: %s' % partname)
             report.append('  * Environment: %s' % environ_name)
             report.append('  * Stage directory: %s' % check.stagedir)
+            if check.job:
+                report.append('  * Nodes: %s' % check.job._nodelist)
```

Comment:
You should use the public attribute, `nodelist`, not `_nodelist`.

Reply:
Also you should format the list with sth like the following: `','.join(check.job.nodelist)`.